Source/ACE.Server/Factories/LootGenerationFactory_Melee.cs

```diff
@@ -318,40 +269,34 @@ private static void MutateMeleeWeapon(WorldObject wo, TreasureDeath profile, boo
                             break;
                     }
                     break;
-                default:
-                    // Two handed
-                    wieldSkillType = Skill.TwoHandedCombat;
-
+                default: // Two handed
                     damage = GetMeleeMaxDamage(wieldSkillType, wieldDiff, LootWeaponType.Cleaving);
                     damageVariance = GetVariance(wieldSkillType, LootWeaponType.TwoHanded);
 
-                    switch (subtype)
+                    switch (wo.W_AttackType)
                     {
-                        case 0:
-                        case 1:
-                        case 2:
-                            weaponDefense = GetMaxDamageMod(profile.Tier, 20);
-                            weaponOffense = GetMaxDamageMod(profile.Tier, 20);
-                            break;
-                        case 3:
-                        case 4:
-                        case 5:
-                        case 6:
-                            weaponDefense = GetMaxDamageMod(profile.Tier, 22);
-                            weaponOffense = GetMaxDamageMod(profile.Tier, 18);
-                            break;
-                        case 7:
-                            weaponDefense = GetMaxDamageMod(profile.Tier, 18);
-                            weaponOffense = GetMaxDamageMod(profile.Tier, 22);
-                            break;
-                        case 8:
-                        case 9:
-                        case 10:
-                        default:
+                        case AttackType.Thrust: // Assagai | Pike | Corsesca | Magari Yari
                             weaponDefense = GetMaxDamageMod(profile.Tier, 15);
                             weaponOffense = GetMaxDamageMod(profile.Tier, 25);
                             damage = GetMeleeMaxDamage(wieldSkillType, wieldDiff, LootWeaponType.Spears);
                             break;
+                        case AttackType.Slash:
+                            // Great Star Mace | Quadrelle | Khanda-handled Mace | Tetsubo
+                            weaponDefense = GetMaxDamageMod(profile.Tier, 22);
+                            weaponOffense = GetMaxDamageMod(profile.Tier, 18);
+
+                            if ((wo.Name.Any() != false) && (wo.Name.Contains("Nodachi") || wo.Name.Contains("Shashqa") || wo.Name.Contains("Spadone")))
```

Comment:
string search is slow, and logic based on item names is very iffy, and will very likely run into edge cases and other bugs here. there has to be better data to drive this logic than item names